trieste/models/model_interfaces.py

```diff
@@ -15,18 +15,28 @@
 from typing import Callable, Dict, Iterable, Optional, Tuple, Union, Any
 
 import gpflow
+from dataclasses import dataclass
 from gpflow.models import GPModel, GPR, SGPR, VGP, SVGP
 import numpy as np
 import tensorflow as tf
 
+from .optimizer import Optimizer
 from .. import utils
 from ..datasets import Dataset
 from ..type import ObserverEvaluations, QueryPoints, TensorType
 
 
+@dataclass
 class ModelInterface(ABC):
     """ A trainable probabilistic model. """
 
+    @abstractmethod
+    def optimize(self, dataset: Dataset) -> None:
+        """
+        Optimize the model.
+        """
+        raise NotImplementedError
```

Comment:
i think the fact that `optimize` is called immediately after `update` and has the same signature is even more reason to merge the two. Users can always define `update` and `optimize` in their own implementations, but wouldn't have to

Reply:
dicsussed online